src/main/java/com/treasure/hunt/jts/geom/HalfPlane.java

```diff
@@ -194,7 +155,12 @@ private Polygon toPolygon(CanvasBoundary boundary) {
         return null;
     }
 
-    public boolean equals(HalfPlane other) {
-        return p0.equals2D(other.p0) && p1.equals2D(other.p1);
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof HalfPlane) {
+            HalfPlane other = (HalfPlane) o;
+            return p0.equals2D(other.p0) && p1.equals2D(other.p1);
+        }
+        return false;
```

Comment:
u could use loombok for that

Reply:
also because the hashcode methosd should then also be adapted

Reply:
Added `@EqualsAndHashCode(callSuper = false)` with param because of the warning

`Generating equals/hashCode implementation but without a call to superclass, even though this class does not extend java.lang.Object. If this is intentional, add '@EqualsAndHashCode(callSuper=false)' to your type.`

Reply:
You know tho that the annotation you have know is different from what you had before if you do not specify what fields are supposed to be compared. idk or maybe just go back to your custom implementation.